app/src/main/java/org/hwyl/sexytopo/control/table/Form.java

```diff
@@ -0,0 +1,66 @@
+package org.hwyl.sexytopo.control.table;
+
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+
+abstract public class Form {
+    static class TextViewValidationTrigger implements TextWatcher {
+        private final Form form;
+
+        TextViewValidationTrigger(Form form) {
+            this.form = form;
+        }
+
+        @Override
+        public void beforeTextChanged(CharSequence charSequence, int start, int before, int count) {}
+
+        @Override
+        public void onTextChanged(CharSequence charSequence, int start, int count, int after) {}
+
+        @Override
+        public void afterTextChanged(Editable editable) {
+            this.form.validate();
+        }
+    }
+
+    interface OnDidValidateCallback {
```

Comment:
Looks strange to not have a public/private/protected keyword on member fields, constructors, nested classes etc. You can skip it but I've known lots of Java coders who have no idea what that does.